awx/main/analytics/collectors.py

```diff
@@ -282,32 +283,43 @@ def _copy_table(table, query, path):
 
 @register('events_table', '1.2', format='csv', description=_('Automation task records'), expensive=True)
 def events_table(since, full_path, until, **kwargs):
-    events_query = '''COPY (SELECT main_jobevent.id, 
+    # The events_table collector will no longer respect the value of
+    # `since`, as that would potentially cause gaps in the output
+    # under circumstances where continuous output is desired.
+
+    truncate = now() - timedelta(weeks=4)
+    truncate_pk = models.JobEvent.objects.filter(created__lt=truncate).aggregate(Max('pk'))['pk__max']
```

Comment:
So when this lands, we will automatically send 4 weeks of data regardless?

Reply:
No.  The caller is still doing the 4-hour time slicing, which is being passed in here as `until`.  This is getting converted into a relevant pk value on line 294, as `final_pk`.  Still just 4 hours at a time.

Reply:
What the above line accomplishes is making sure that if our starting point is more than 4 weeks in the past, it will bring it forward to the 4-week mark.  It's needed because since we are dealing with pk values instead of timestamps, the equivalent truncate-at-4-weeks logic back in the caller doesn't do anything with this collector.